internal/command/jsonstate/state.go

```diff
@@ -496,14 +496,34 @@ func marshalResources(resources map[string]*states.Resource, module addrs.Module
 }
 
 func SensitiveAsBool(val cty.Value) cty.Value {
+	return SensitiveAsBoolWithPathValueMarks(val, nil)
+}
+
+func SensitiveAsBoolWithPathValueMarks(val cty.Value, pvms []cty.PathValueMarks) cty.Value {
+	var path cty.Path
+	sensitiveMarks := make([]cty.PathValueMarks, 0)
+	for _, pvm := range pvms {
+		if _, ok := pvm.Marks[marks.Sensitive]; ok {
+			sensitiveMarks = append(sensitiveMarks, pvm)
+		}
+	}
+	return sensitiveAsBoolWithPathValueMarks(val, path, sensitiveMarks)
```

Comment:
tiny style suggestion: it may be more obvious to the reader if we pass an empty `cty.Path` here (with no variable)

Reply:
updated